src/common/components/Exercises.tsx

```diff
@@ -52,62 +52,79 @@ export const Exercises: React.FC<ExercisesProps> = ({ title, defaultDuration, ty
     });
   };
 
+  const loadingText = (
+    <Text as="span" fontSize="2xl" fontWeight={"normal"} ml={"20px"} visibility={isFetching ? "visible" : "hidden"}>
+      Loading...
+    </Text>
+  );
+
   return (
-    <Box mt={"3%"}>
-      <VStack align="start" spacing={4}>
-        <Text fontSize="2xl" fontWeight={"normal"} >{title} { (defaultDuration != 0 && defaultDuration) && (<Text as="span" color={"blue"}>{defaultDuration} days</Text>)} {(isFetching) && <Text as="span" fontSize="2xl" fontWeight={"normal"} ml={"20px"}> Loading... </Text>}</Text>
+    <Box mt={"3%"} width={"100%"}>
+      <VStack align="start" spacing={4} width="100%">
+        <Text fontSize="2xl" fontWeight={"normal"}>
+          {title}
+          {(defaultDuration != 0 && defaultDuration) && (
+            <Text as="span" color={"blue"}>{defaultDuration} days</Text>
+          )}
+          {loadingText}
```

Comment:
I think we should do something like this

{isFetching && <Text as="span" fontSize="2xl" fontWeight={"normal"} ml={"20px"} >
      Loading...
    </Text>}

Reply:
This will allow us to have all of JSX at one place improving the readability, what do you think?